x-pack/plugins/encrypted_saved_objects/server/crypto/encryption_error.ts

```diff
@@ -16,4 +25,8 @@ export class EncryptionError extends Error {
     // https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#extending-built-ins-like-error-array-and-map-may-no-longer-work
     Object.setPrototypeOf(this, EncryptionError.prototype);
   }
+
+  toJSON() {
+    return { message: this.message };
```

Comment:
__note:__ I dug into how we generate errors in SO client and we basically have two choices:

* Error [extracted from the payload](https://github.com/elastic/kibana/blob/592e23496f9b7072f252946502ace4692fb71ab8/src/core/server/saved_objects/service/lib/repository.ts#L294) that is HTTP-centric (e.g. includes status code)

* Generic error that [includes only error message](https://github.com/elastic/kibana/blob/592e23496f9b7072f252946502ace4692fb71ab8/src/core/server/saved_objects/service/lib/repository.ts#L1438-L1440)

In our case error has nothing to do with HTTP, status codes and such so having just error message sounds reasonable to me.

The `error` property of the saved object will be raw `EncryptionError` that will allow consumers to use `isEncryptionError` API we expose and will be "converted" to `{ message: string }` when serialized within REST API response.